### PR TITLE
Navbar

### LINKED ISSUE #31: Navbar When Browsing
**Description of the issue**
We need to have a navbar component so that we can show off the logo, info of a user, and some quick navigation. This issue targets the *frontend*. ideally we want a user to be able to sign out using the nav bar. In the future we can make it so that you can also sign in using the navbar if we ever have a unauthenticated view of our app. 

**Files that need to be updated**
We would need to add some components in `src/components/` to be able to export the nav bar onto different pages.

**Additional context**
The navbar is most likely going to be used on every page. we will assume that a user has already signed in to be viewing a navbar, this could change in the future.

### PR DESCRIPTION
**Pull request by barrioskevin**
*Please complete the checklist so the group can quickly understand your PR.*

### Category
- [X] **Feature**
- [ ] **Bug Fix**
- [ ] **Testing**
- [ ] **Documentation**
- [ ] **Other**

### Domain
- [X] **Frontend**
- [ ] **Backend**
- [ ] **Both**

### Description
This pull request is an attempt to bring a navbar component into our project. The navbar will live at the top of every page and will provide user actions and quick navigations. The navbar I am submitting on this pull request is a work in progress and has not achieved the full functionality. The reason I'm making a pull request is in order to get feedback from the rest of the group and maybe you guys can see how the navbar works and add in the routes for the different pages. This navbar will at the very least be able to make our prototype look more complete.

### Related Issue(s)
Closes #31 . New issues should be open later to target any specific issues with the navbar.

### New Additions
`frontend/public/navbar/logo/*` - added in some logos for rebelhub (3 pngs)
`frontend/public/navbar/icons/*` - added in icons used in the navbar (only useful for dark theme, 3 pngs)
`frontend/src/components/navbar/*` - added in navbar component and a corresponding css

### Modifications
No modifications this is all new content.

### New Dependencies
No new dependencies.

### Visuals
I left some of the actual page in the screenshots so you can see how it would look on a page. I apologize if the screenshots didn't come out that well. The logo is also easily swappable if you guys prefer the logo with the rebel.
#### Normal view of navbar
image
#### Light theme view of navbar
image
#### Skinny view of navbar
image

### Architecture of Your Changes
The navbar component is a nav element with 3 buttons, a search bar, and an account component that is a mini account button where the user can see their avatar and username. 

The account component takes in some props. It takes in a darkTheme value (so we can simulate how the same logic might apply for the navbar component itself if it were to get a darktheme value from somewhere else) and also a username value. I added the username value that way when we eventually pull the data from the user it will show their name. (it just occured to me I can probably make that call myself right now in order to display the actual username but let me know if i should).

There is a home button which is just the logo, messages button, settings button, and account button. 

All buttons have their functions for handling presses, just no routes have been applied.

The way I have it setup right now when you press the settings button it will toggle between light and dark mode. I didn't fully implement a dark view and light view since I wasn't completely sure if we were going to have that expand across the whole app. If we stick with dark mode for the time being this navbar should look fine.  

Also it's worth noting that this navbar component internally handles any overlapping content that could be hiding behind it. I have a div just for extra space and it's controlled within the component and css. 

### Usage and Integration
For using the navbar just do 
```javascript
import RebelHubNavBar from '@/components/navbar/RebelHubNavBar'
```
Then you can render the `<RebelHubNavBar />` component on any page. Right now there isn't really much integration with the rest of the project but as we build out the remaining pages we can easily edit the button functions to route to the new pages on press.

### Testing
- **Testing Approach**: I mainly used the logs from the frontend server to see if I was getting the images from the `public/` directory. I also used the developer console in order to see my logs on button presses.
- **Edge Cases**: We need to account for small width screens, at the moment i stacked all the icons but this should be updated in the future. We also need to be aware of any state used in the navbar that might expand across the entire app that way we could just pass in the global state to the navbar. Right now it's not an issue since the navbar component manages a darkTheme on it's own.

### Mentions
I was also thinking that maybe we can replace either settings or messages with a calendar button since that is something we already have. If that's the case then @nnguyen144 should be aware of this PR so he can add in the route to the calendar page. @leed85 and @courtmcgee2197 should also be aware of this PR as they can add in the routes for the account component. (take user to their profile page, sign out option)

Let me know if there's anything that I should get added before this can be merged and you guys can start testing out the feature and adding on it. Thank you for your time, 
- Kevin Barrios

### CODE REVIEW

**Comment by thecatfour** [comment]
I think it might be a good idea to move the AccountComponent into its own file. This way, we can reuse it for things like comments where people can click on the component to view the commenter's profile.

**Comment by leed85** [comment]
I think this looks great, the only issue (which im pretty sure its just for now) is the logos disappear in lightmode. I was also maybe thinking for settings and messages, we move it under the profile (i.e. you press profile and a dropdown menu appears with those options). Also  for darkmode/lightmode should we allow users to easily change between the two with the navbar, or put that in options? Overall I think it looks good.

File: frontend/src/components/navbar/AccountButton.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import styles from './AccountButton.module.css';
+
+
+const avatarIconPath = "/navbar/icons/avatar.png" // path for the avatar icon.
+
+//takes in a props{username, darkTheme} 
+//username can be changed to be the actual user id. 
+
+
+const AccountButton = (props) => {
```
Comment (thecatfour) [inline frontend/src/components/navbar/AccountButton.js:14]: I believe you can unpack the props using code like the following:
```
const AccountButton = ({ username, darkTheme }) => {
     const [displayName, setDisplayName] = useState(username);
     ...
```
This makes it easier to understand what a component needs to be passed. 
ChatGPT also agrees: https://chatgpt.com/share/6723c8f5-7e9c-8004-bac9-2329873d84b5

**Review by thecatfour — DISMISSED** [review]

**Review by leed85 — APPROVED** [review]

File: frontend/public/navbar/logo/logo_dark_alt.png
Comment (nnguyen144) [inline frontend/public/navbar/logo/logo_dark_alt.png:1]: This logo should probably not be used as it is too similar to pre-existing logos for other products.

File: frontend/public/navbar/logo/logo_light_alt.png
Comment (nnguyen144) [inline frontend/public/navbar/logo/logo_light_alt.png:1]: This logo should probably not be used as it is too similar to pre-existing logos for other products.

File: frontend/src/components/navbar/AccountButton.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import React, { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import styles from './AccountButton.module.css';
+
+
+const avatarIconPath = "/navbar/icons/avatar.png" // path for the avatar icon.
+
+//takes in a props{username, darkTheme} 
+//username can be changed to be the actual user id.
```
Comment (nnguyen144) [inline frontend/src/components/navbar/AccountButton.js:11]: user id should be more adequate then using username.

**Comment by nnguyen144** [comment]
I think for now, the Navbar looks good! There are some issues that will be fixed later on (such as a better logo and no router pushes implemented) but for now this PR can be merged.

**Review by nnguyen144 — APPROVED** [review]
I think for now, the Navbar looks good! There are some issues that will be fixed later on (such as a better logo, better light mode/dark mode appearance, and no router pushes implemented) but as long as the Navigation bar is able to be seen on the top of a page with correct fixed positioning, this PR can be merged so we can work on our pages.

**Pull request merged**